File: Exercise_29.py

# Print the distance between the first 100 prime numbers
from Exercise_16 import is_prime

def distance_between_prime(n):
    last_prime = 2
    i = last_prime + 1
    counter = 1
    while counter < n:
        if is_prime(i):
            print(f"{i - last_prime} \t {i} - {last_prime} ")
            last_prime = i
            counter += 1
        i += 1


distance_between_prime(100)

File: Randy-JRA/Exercise_29.cpp
//Challenge 29 is to Print the distance between the first 100 prime numbers

#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(int n) {
    if (n <= 1) {
        return false;
    }
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int count = 0;
    int prev_prime = 0;
    for (int i = 2; count < 100; i++) {
        if (is_prime(i)) {
            if (prev_prime != 0) {
                cout << i - prev_prime << endl;
            }
            prev_prime = i;
            count++;
        }
    }
    return 0;
}
 
